src/platform/backends/shared/base_virtual_machine.cpp

```diff
@@ -515,6 +515,11 @@ void BaseVirtualMachine::restore_rollback_helper(const Path& head_path,
 }
 
 void BaseVirtualMachine::restore_snapshot(const std::string& name, VMSpecs& specs)
+{
+    restore_snapshot_impl(name, specs);
+}
+
+void BaseVirtualMachine::restore_snapshot_impl(const std::string& name, VMSpecs& specs)
```

Comment:
Why did you choose to add this? I understand that extra interfaces aren't supported on all backends, but on those that don't support it, they would just be empty, so the generic handling would apply (with most likely insignificant detriment to perf), no?

Reply:
Hey @ricab, I did this because QEMU backend does exactly what this function does, plus modifying an internal data structure. It was the easiest way of reusing the code in other backend.

Reply:
Hey @luis4a0, I still don't understand this. You could just call `BaseVM::restore` from `QemuVM::restore`, you don't need a separate impl method for that. However, I don't think you need even that. You can just update extra interfaces together with other `desc` fields [in the snapshot](https://github.com/canonical/multipass/blob/acf850a59f5bee58ae0992fe4f069ed101e37e31/src/platform/backends/qemu/qemu_snapshot.cpp#L101).

Reply:
In fact you already update the `desc` in the `QemuSnapshot::apply`, which is [called from restore](https://github.com/canonical/multipass/blob/922646dc81ee127032ba0501bcce36e41138cd1d/src/platform/backends/shared/base_virtual_machine.cpp#L542).

Reply:
Hey @ricab, thanks! Right, I removed the implementation of `QemuVirtualMachine::restore_snapshot()`. I am currently testing to see if I'm not forgetting something. Will let you know when I'm OK with the code.